src/azure-cli/setup.py

```diff
@@ -144,6 +144,7 @@
     'PyNaCl~=1.5.0',
     'scp~=0.13.2',
     'semver==2.13.0',
+    'setuptools',
```

Comment:
@bebound Hi! I don't believe `install_requires` is the correct place to add `setuptools` - instead you should use `setup_requires`, since the former isn't installed before the rest of the other packages. Also note that even after the get-pip changes, pip still injects setuptools/wheel via the isolated build environments, so it may not be necessary for you to add `setuptools` at all. Lastly, you may want to consider migrating to `pyproject.toml` instead of `setup.py` and declaring an explicit build backend per [PEP 518](https://peps.python.org/pep-0518/).

Reply:
The reason why this PR seeks to add setuptools to run-time dependencies is the use of `pkg_resources` to declare an old-style namespace package. This, of course, is something that should be eliminate entirely instead of adding `setuptools` to run-time dependencies.

Reply:
Here is the search result. I'm not sure if any extension uses it.
```
azure/multiapi/storagev2/queue/__init__.py
1:__import__('pkg_resources').declare_namespace(__name__)

azure/multiapi/storagev2/blob/__init__.py
1:__import__('pkg_resources').declare_namespace(__name__)

azure/multiapi/storagev2/filedatalake/__init__.py
1:__import__('pkg_resources').declare_namespace(__name__)

azure/multiapi/storagev2/fileshare/__init__.py
1:__import__('pkg_resources').declare_namespace(__name__)

azure/multiapi/cosmosdb/v2017_04_17/__init__.py
1:__import__('pkg_resources').declare_namespace(__name__)

azure/cli/command_modules/serviceconnector/_utils.py
382:    import pkg_resources
383:    installed_packages = pkg_resources.working_set
```

Reply:
Perhaps the storage-multiapi package should be fixed first. There's no point in it supporting Python 2.7 anymore, as its only downstream package (this one) requires Python 3.8 or later.